proposals/params-collections.md

```diff
@@ -144,8 +142,105 @@ In case the parameter type sequences `{P₁, P₂, ..., Pᵥ}` and `{Q₁, Q₂,
+> - **both params collections are not *span_type*s, and an implicit conversion exists from params collection of `Mᵢ` to params collection of `Mₑ`** 
+
+it is applicable to arrays and, therefore, performing the tie-break earlier will introduce a behavior change for existing scenarios.
+
+For example:
+``` C#
+class Program
+{
+    static void Main()
+    {
+        Test(1);
+    }
+
+    static void Test(in int x, params C2[] y) {} // There is an implicit conversion from `C2[]` to `C1[]`
+    static void Test(int x, params C1[] y) {} // Better candidate because of "better parameter-passing choice"
+}
+
+class C1 {}
+class C2 : C1 {}
+```
+
+If any of the previous tie-breaking rules apply (including the "better arguments conversions" rule), the overload resolution result
+can be different by comparison to the case when an explicit collection expression is used as an argument instead.
+
+For example:
+``` C#
+class Program
+{
+    static void Test1()
+    {
+        M1(['1', '2', '3']); // Span overload is used
+        M1('1', '2', '3');   // String overload is used because `char` is an exact match
+    }
+
+    static void M1(params string value) {}
+    static void M1(params System.ReadOnlySpan<MyChar> value) {}
+
+    class MyChar
+    {
+        private readonly int _i;
+        public MyChar(int i) { _i = i; }
+        public static implicit operator MyChar(int i) => new MyChar(i);
+        public static implicit operator char(MyChar c) => (char)c._i;
+    }
+
+    static void Test2()
+    {
+        M2([1]); // Span overload is used
+        M2(1);   // Array overload is used, not generic
+    }
+
+    static void M2<T>(params System.Span<T> y){}
+    static void M2(params int[] y){}
+
+    static void Test3()
+    {
+        M3("3", ["4"]); // Span overload is used, better on the first argument conversion, none is better on the second
+        M3("3", "4");   // Ambiguity, better-ness of argument conversions goes in opposite directions.
+                        // Since parameter types are different ("object, string" vs. "string, object"), tie-breaking rules do not apply
+    }
+
+    static void M3(object x, params string[] y) {}
+    static void M3(string x, params Span<object> y) {}
+}
+```
+
+However, our primary concern are scenarios where overloads differ only by params collection type,
+but the collection types have the same element type. The behavior should be consistent with 
+explicit collection expressions for these cases.
+
+
+The "**if the same set of arguments corresponds to the collection elements for both methods**" condition is important for scenarios like:
+``` C#
+class Program
+{
+    static void Main()
+    {
+        Test(x: 1, y: 2); // Ambiguous
+    }
+
+    static void Test(int x, params System.ReadOnlySpan<int> y) {}
+    static void Test(int y, params System.Span<int> x) {}
+}
+```
+
+It doesn't feel reasonable to "compare" collections that are built from different elements.
```

Comment:
I think I can imagine this occurring when the methods have different number of parameters but both have a `params` parameter. Still, an example may help here.

Reply:
> I think I can imagine this occurring when the methods have different number of parameters but both have a `params` parameter. Still, an example may help here.

The different number of parameters is handled by a different pre-existing rule.